### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by vikramvijayan** [NeurodataWithoutBorders/matnwb#430]
### What happened?

I need to make a timeseries with only 1 entry (it is a sparse behavioral annotation of my data). I make both the data and timestamp a single entry and MatNWB makes the file fine, but I get an error when running nwbinspector.

Please let me know what is the best option for me to include this behavioral annotation. I cannot upload my data to DANDI because of this error (the NWB file itself looks fine other than this error).

### Steps to Reproduce

```matlab
spatial_series_ts12 = types.core.TimeSeries( ...
            'data', 12.3, ...
            'description', 'XYZ', ...
            'data_unit', 'time', ...
            'timestamps', 12.3);
```

### Error Message

```matlab
TypeError: TimeSeries.init: incorrect type for 'data' (got 'float64', expected 'ndarray, list, tuple, Dataset, StrDataset, HDMFDataset, AbstractDataChunkIterator, DataIO or TimeSeries'), incorrect type for 'timestamps' (got 'float64', expected 'ndarray, list, tuple, Dataset, StrDataset, HDMFDataset, AbstractDataChunkIterator, DataIO or TimeSeries')"
```

### Operating System

Windows

### Matlab Version

2021A

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/neurodatawithoutborders/matnwb/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/matnwb/issues)?

**Comment by oruebel** [comment]
Based on the error, it seems that the data is stored as single scalar value, rather than as an array. Instead of providing the data as a single value, I believe you need to input it as an array instead. I think the following should work: 

```
spatial_series_ts12 = types.core.TimeSeries( ...
            'data', [12.3, ], ...
            'description', 'XYZ', ...
            'data_unit', 'time', ...
            'timestamps', [12.3, ]);
```

Quick question about the data. In the description you mention that this is an 'XYZ' location but the data is a scalar. If this is indeed a spatial series then https://neurodatawithoutborders.github.io/matnwb/doc/+types/+core/SpatialSeries.html could potentially be a more appropriate type. Also, I notices that the timestamps have the same value as the data, is this expected?

**Comment by CodyCBakerPhD** [comment]
@oruebel I think problem at the heart of it is MATLAB doesn't seem to distinguish the data type at all between the commands

```matlab
a=1
b=[1]
c=[1,]

a==b
> 1 (logical)

b==c
> 1 (logical)
```

only if there is, in fact, more than one item present in the array.

**Comment by vikramvijayan** [comment]
@oruebel see @CodyCBakerPhD 's comment.

Also, "XYZ" was just meant to be a placeholder for an actual description (not coordinate information). Sorry for that confusion.

I made the 'data' and 'timestamp' the same because it is simple a behavioral annotation (when an animal laid an egg). Alternatively I could put data = 1 and timestamp = 12.3s. I'm open to both options but that won't fix the initial problem at hand.

**Comment by oruebel** [comment]
> I think problem at the heart of it is MATLAB doesn't seem to distinguish the data type at all between the commands

Thanks for the clarification. I did not know that. That seems to be a question for @lawrence-mbf

**Comment by lawrence-mbf** [comment]
This is fixable, it just has to do with how datasets know they must always be non-scalar as derived from the schema.

Is the case below the only time in the schema where a scalar dataset would be allowed?

```
shape:
- - 1
- - null 
  - null
... etc
```

**Comment by vikramvijayan** [comment]
Thanks! Does this mean that once the commit is merged I can update matNWB and be ready to go?

**Comment by lawrence-mbf** [comment]
Unfortunately, you'd have to recreate the file but otherwise yes.

**Comment by vikramvijayan** [comment]
No problem at all! I'll monitor to see when the code has been updated!

**Comment by bendichter** [comment]
@lawrence-mbf it sounds like the solution you are planning is ideal- defer to the schema to see if the dataset should be a vector. If that it difficult, another option could be to use `DataPipe`, but it seems like that would be far less user-friendly than what you are thinking.

**Comment by lawrence-mbf** [comment]
@vikramvijayan Sorry for letting this fall through. Would it be possible for you to check if this resolves the warning? The code is currently still in the PR.

**Comment by vikramvijayan** [comment]
Thanks @lawrence-mbf ! How would I go about downloading the code that is in PR? Downloading a .zip would be the easiest and I can take it from there.

**Comment by lawrence-mbf** [comment]
@vikramvijayan You should be able to download from this URL: https://github.com/NeurodataWithoutBorders/matnwb/tree/430-force-dataset-array-shape.

**Comment by vikramvijayan** [comment]
Thanks @lawrence-mbf , it appears the updated code works!
